pkg/storage/etcd3/watcher.go

```diff
@@ -276,6 +281,11 @@ func parseError(err error) *watch.Event {
 }
 
 func (wc *watchChan) sendError(err error) {
+	// Context.canceled is an expected behavior.
+	// We should just stop all goroutines in watchChan without returning error.
+	if err == context.Canceled {
```

Comment:
probably 
1. type assert on grpc error and then compare the code. 
2. if not grpc, compare with context.Canceled.

I think this is kind of etcd/clientv3 issue. We should not leak gRPC error. But this is P1 for k8s, so let's fix this in k8s.

/cc @heyitsanthony